tests/rptest/utils/si_utils.py

```diff
@@ -1049,7 +1050,15 @@ def _do_listing(self):
                 self._state.segment_objects += 1
                 if self._scan_segments:
                     spc = parse_s3_segment_path(o.key)
-                    self._add_segment_metadata(o.key, spc)
+                    try:
+                        self._add_segment_metadata(o.key, spc)
+                    except ClientError as err:
+                        # The segment was listed by ListObjectV2 request
+                        # and deleted by Redpanda concurrently.
+                        # We don't expect this to happen with the manifests
+                        # so this error is only handled in case of segments
+                        if err['Error']['Code'] == 'NoSuchKey':
+                            self._state.ignored_objects += 1
```

Comment:
so if it isn't NoSuckKey should we rethrow it?

Reply:
not really
this means that the code is racing with GC in redpanda, this is not an error